src/signature.rs

```diff
@@ -105,6 +108,19 @@ impl fmt::Display for Signature {
     }
 }
 
+impl FromStr for Signature {
```

Comment:
Bytes64 already implements [FromStr](https://docs.rs/fuel-types/0.3.1/fuel_types/struct.Bytes64.html#impl-FromStr)

We shouldn't replace the underlying implementation - instead, we should reuse it
```suggestion
impl From<Bytes64> for Signature {
    fn from(b: Bytes64) -> Self {
        Self(b)
    }
}

impl From<Signature> for Bytes64 {
    fn from(s: Signature) -> Self {
        s.0
    }
}

impl FromStr for Signature {
    type Err = Error;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        Bytes64::from_str(s)
            .map_err(|_| Error::InvalidSignature)
            .map(|s| s.into())
    }
}
```